sciris/sc_parallel.py

```diff
@@ -64,63 +64,117 @@ def loadbalancer(maxload=None, index=None, interval=None, maxtime=None, label=No
-            xy = [x+i*pl.rand(1000), y+i*pl.randn(1000)]
+            xy = [x+i*pl.rand(100), y+i*pl.randn(100)]
             return xy
-        n = 5
-        xy = parallelize(myfunc, kwargs={'x':3, 'y':8}, iterarg=n, itervar='i', maxload=0.8)
-        for i in range(n):
-            pl.scatter(xy[i][0], xy[i][1])
+        xylist = sc.parallelize(myfunc, kwargs={'x':3, 'y':8}, iterarg=range(5), maxload=0.8)
+        for xy in xylist:
+            pl.scatter(xy[0], xy[1])
     
     Version: 2018nov01
     '''
     # Handle maxload
     if ncpus is None and maxload is None:
         ncpus = mp.cpu_count()
     
-    # Handle iterarg
-    if not(ut.isiterable(iterarg)):
-        try:
-            iterarg = np.arange(iterarg)
-        except Exception as E:
-            errormsg = 'Could not understand iterarg "%s": not iterable and not an integer: %s' % (iterarg, str(E))
+    # Handle iterarg and iterkwargs
+    niters = 0
+    if iterarg is not None and iterkwargs is not None:
+        errormsg = 'You can only use one of iterarg or iterkwargs as your iterable, not both'
+        raise Exception(errormsg)
+    if iterarg is not None:
+        if not(ut.isiterable(iterarg)):
+            try:
+                iterarg = np.arange(iterarg)
+            except Exception as E:
+                errormsg = 'Could not understand iterarg "%s": not iterable and not an integer: %s' % (iterarg, str(E))
+                raise Exception(errormsg)
+        niters = len(iterarg)
+    if iterkwargs is not None: # Check that iterkwargs has the right format
+        if not isinstance(iterkwargs, dict):
+            errormsg = 'iterkwargs must be a dict or None, not %s' % type(iterkwargs)
             raise Exception(errormsg)
+        for key,val in iterkwargs.items():
+            if not ut.isiterable(val):
+                errormsg = 'iterkwargs entries must be iterable, not %s' % type(val)
+                raise Exception(errormsg)
+            if not niters:
+                niters = len(val)
+            else:
+                if len(val) != niters:
+                    errormsg = 'All iterkwargs iterables must be the same length, not %s vs. %s' % (niters, len(val))
+                    raise Exception(errormsg)
     
     # Construct argument list
     argslist = []
-    niters = len(iterarg)
-    for index,iterval in enumerate(iterarg):
-        taskargs = TaskArgs(func, index, iterval, itervar, args, kwargs, maxload)
+    for index in range(niters):
+        if iterarg is None:
+            iterval = None
+        else:
+            iterval = iterarg[index]
+        if iterkwargs is None:
+            iterdict = None
+        else:
+            iterdict = {}
+            for key,val in iterkwargs.items():
+                iterdict[key] = val[index]
+        taskargs = TaskArgs(func, index, iterval, iterdict, args, kwargs, maxload)
         argslist.append(taskargs)
         
     # Decide how to run -- not load-balanced, use map
```

Comment:
So I realized on closer inspection it was actually just the comment that was confusing, because `maxload` always gets passed into `parallel_task` via `TaskArgs` so even though the comment says 'not load-balanced' it still might be depending on `maxload` :)